File: sdk-py/getpatter/tts/elevenlabs.py
"""ElevenLabs TTS for Patter pipeline mode."""

from __future__ import annotations

import os

from getpatter.providers.elevenlabs_tts import ElevenLabsTTS as _ElevenLabsTTS

__all__ = ["TTS"]


class TTS(_ElevenLabsTTS):
    """ElevenLabs streaming TTS.

    Example::

        from getpatter.tts import elevenlabs

        tts = elevenlabs.TTS()              # reads ELEVENLABS_API_KEY
        tts = elevenlabs.TTS(api_key="...", voice_id="EXAVITQu4vr4xnSDxMaL")
    """

    def __init__(
        self,
        api_key: str | None = None,
        *,
        voice_id: str = "EXAVITQu4vr4xnSDxMaL",
        model_id: str = "eleven_flash_v2_5",
        output_format: str = "pcm_16000",
    ) -> None:
        key = api_key or os.environ.get("ELEVENLABS_API_KEY")
        if not key:
            raise ValueError(
                "ElevenLabs TTS requires an api_key. Pass api_key='...' or "
                "set ELEVENLABS_API_KEY in the environment."
            )
        super().__init__(
            api_key=key,
            voice_id=voice_id,
            model_id=model_id,
            output_format=output_format,
        )

File: sdk-ts/src/tts/elevenlabs.ts
/** ElevenLabs TTS for Patter pipeline mode. */
import { ElevenLabsTTS as _ElevenLabsTTS, type ElevenLabsModel } from "../providers/elevenlabs-tts";

export type { ElevenLabsModel };

export interface ElevenLabsTTSOptions {
  /** API key. Falls back to ELEVENLABS_API_KEY env var when omitted. */
  apiKey?: string;
  voiceId?: string;
  /**
   * ElevenLabs voice model ID. Default is ``eleven_flash_v2_5`` (lowest TTFT).
   * Pass ``eleven_v3`` for highest quality, or any string for forward-compat.
   */
  modelId?: ElevenLabsModel | string;
  outputFormat?: string;
}

/**
 * ElevenLabs TTS.
 *
 * @example
 * ```ts
 * import * as elevenlabs from "getpatter/tts/elevenlabs";
 * const tts = new elevenlabs.TTS();              // reads ELEVENLABS_API_KEY
 * const tts = new elevenlabs.TTS({ apiKey: "...", voiceId: "rachel" });
 * ```
 */
export class TTS extends _ElevenLabsTTS {
  static readonly providerKey = "elevenlabs";
  constructor(opts: ElevenLabsTTSOptions = {}) {
    const key = opts.apiKey ?? process.env.ELEVENLABS_API_KEY;
    if (!key) {
      throw new Error(
        "ElevenLabs TTS requires an apiKey. Pass { apiKey: '...' } or " +
          "set ELEVENLABS_API_KEY in the environment.",
      );
    }
    super(
      key,
      opts.voiceId ?? "EXAVITQu4vr4xnSDxMaL",
      opts.modelId ?? "eleven_flash_v2_5",
      opts.outputFormat ?? "pcm_16000",
    );
  }
}
